apps/liteprotocoltester/statistics.nim

```diff
@@ -42,95 +51,143 @@ proc init*(T: type Statistics, expectedMessageCount: int = 1000): T =
   result.helper.prevIndex = 0
   result.helper.maxIndex = 0
   result.helper.seenIndices.init(expectedMessageCount)
-  result.minLatency = nanos(0)
-  result.maxLatency = nanos(0)
-  result.cummulativeLatency = nanos(0)
+  result.received = initTable[uint32, MessageInfo](expectedMessageCount)
   return result
 
-proc addMessage*(self: var Statistics, msg: ProtocolTesterMessage) =
+proc addMessage*(self: var Statistics, msg: ProtocolTesterMessage, msgHash: string) =
   if self.allMessageCount == 0:
     self.allMessageCount = msg.count
+    self.firstReceivedIdx = msg.index
   elif self.allMessageCount != msg.count:
-    warn "Message count mismatch at message",
+    error "Message count mismatch at message",
       index = msg.index, expected = self.allMessageCount, got = msg.count
 
-  if not self.helper.seenIndices.contains(msg.index):
-    self.helper.seenIndices.incl(msg.index)
-  else:
+  let currentArrived: MessageInfo = (
+    msg: msg,
+    info: ArrivalInfo(
+      arrivedAt: Moment.now(),
```

Comment:
Maybe it doesn't affect to the `liteprotocoltester` tool but the most standard way that we can follow to get "now" I think is 

```nim
proc getNowInNanosecondTime*(): Timestamp =
  return getNanosecondTime(nowInUnixFloat())
```
(waku/waku_core/time.nim)